src/main/java/net/minecraftforge/client/IArmorOverlay.java

```diff
@@ -0,0 +1,120 @@
+import net.minecraft.inventory.EntityEquipmentSlot;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.ResourceLocation;
+
+public interface IArmorOverlay
+{
+
+    /**
+     * Vanilla's default armor overlay color value. Uses half the alpha value than the Item color.
+     * */
+    public static final int defaultOverlayColor = 0xAA8040CC;
+
+    /**
+     * Vanilla's default Glint texture.
+     * */
+    public static final ResourceLocation defaultGlint = new ResourceLocation("textures/misc/enchanted_item_glint.png");
+
+    public static final IArmorOverlay VANILLA = new Vanilla();
+
+    /**
+     * Used for overlay coloring when the Armor's effect passes are run.
+     * 
+     * @param stack the ItemStack
+     * @param wearer the Entity in question
+     * @param pass the render pass
+     * @param armorSlot the slot the armor is in
+     * 
+     * @return a hex int (ARGB) used for coloring the effect
+     * */
+    public int getOverlayColor(ItemStack stack, EntityLivingBase wearer, int pass, EntityEquipmentSlot slot);
+
+    /**
+     * Used for effect scaling based on time.
+     * 
+     * @param stack the ItemStack
+     * @param wearer the Entity in question
+     * @param pass the render pass
+     * @param slot the slot the armor is in
+     * @param time the offset time
+     * 
+     * @return a 3 value array of floats representing scaling
+     * */
+    public float[] getOverlayScaleVector(ItemStack stack, EntityLivingBase wearer, int pass, EntityEquipmentSlot slot, float time);
+
+    /**
+     * Used for effect translation based on time.
+     * 
+     * @param stack the ItemStack
+     * @param wearer the Entity in question
+     * @param pass the render pass
+     * @param slot the slot the armor is in
+     * @param time the offset time
+     * 
+     * @return a 3 array of floats representing translation
+     * */
+    public float[] getOverlayTranslationVector(ItemStack stack, EntityLivingBase wearer, int pass, EntityEquipmentSlot slot, float time);
+    /**
+     * Used for effect rotation based on time.
+     * 
+     * @param stack the ItemStack
+     * @param wearer the Entity in question
+     * @param pass the render pass
+     * @param slot the slot the armor is in
+     * @param time the offset time
+     * 
+     * @return a 4 value array of floats representing rotation
+     * */
+    public float[] getOverlayRotationVector(ItemStack stack, EntityLivingBase wearer, int pass, EntityEquipmentSlot slot, float time);
+
+    /**
+     * Used for setting the effect texture at the start of a pass.
+     * 
+     * @param stack the ItemStack in question
+     * @param wearer the Entity in question
+     * 
+     * @return the ResourceLocation of the effect texture.
+     * */
+    public ResourceLocation getOverlayTexture(ItemStack stack, EntityLivingBase wearer);
+    
+    public static class Vanilla implements IArmorOverlay
+    {
+
+        private Vanilla(){}
```

Comment:
This should be protected so modders can subclass it with their own small changes.